yarn-project/circuits.js/src/contract/contract_class.ts

```diff
@@ -16,6 +16,7 @@ export function getContractClassFromArtifact(
   const artifactHash = 'artifactHash' in artifact ? artifact.artifactHash : computeArtifactHash(artifact);
   const publicFunctions: ContractClass['publicFunctions'] = artifact.functions
     .filter(f => f.functionType === FunctionType.OPEN)
+    .sort((a, b) => a.name.localeCompare(b.name))
```

Comment:
Do we (why don't we) need to do the same thing for the private functions below? 
Also, we should probably update the comments in `yarn-project/circuits.js/src/contract/artifact_hash.ts` and `yellow-paper/docs/contract-deployment/classes.md` so this isn't lost.

Reply:
Also, we should probably sort based on the selector and not the name.